websites/J/JoJo's Bizarre Encyclopedia/presence.ts

```diff
@@ -14,150 +19,184 @@ presence.on("UpdateData", () => {
 		)
 			presenceData.state = "Viewing homepage";
-		else if (window.location.pathname.toLowerCase().includes("/index.php")) {
-			const search = new URL(window.location.href).searchParams.get("search"),
-				// Planning to add more actions, so it'll be used multiple times in the future.
-				// eslint-disable-next-line no-one-time-vars/no-one-time-vars
-				action = new URL(window.location.href).searchParams.get("action");
+		else if (pathname.toLowerCase().includes("/index.php")) {
+			const search = searchParams.get("search"),
+				action = searchParams.get("action");
+
+			let title;
+			if (searchParams.get("title"))
+				title = urlToTitle(searchParams.get("title"));
+			else title = "Homepage";
 
-			if (search.length > 0) presenceData.state = `Searching "${search}"`;
+			if (search?.length > 0) presenceData.state = `Searching "${search}"`;
 			else if (action === "history") {
-				let title = new URL(window.location.href).searchParams.get("title");
-				title = urlToTitle(title);
-				presenceData.state = `Viewing history of ${title}`;
+				presenceData.state = `Viewing history: ${title}`;
+				if (showButtons) {
+					presenceData.buttons = [
+						{
+							label: "View Page",
+							url: `${origin}/${titleToUrl(title)}`,
+						},
+					];
+				}
+			} else if (action === "purge") {
+				presenceData.state = `Purging: ${title}`;
+				if (showButtons) {
+					presenceData.buttons = [
+						{
+							label: "View Page",
+							url: `${origin}/${titleToUrl(title)}`,
+						},
+					];
+				}
+			}
+		} else if (pathname.toLowerCase().startsWith("/user:")) {
+			presenceData.state = `Viewing User: ${pathname.match(/\/user:(.*)/i)[1]}`;
+			if (showButtons) {
 				presenceData.buttons = [
 					{
-						label: "View Page",
-						url: `${window.location.origin}/${titleToUrl(title)}`,
+						label: "View User",
+						url: href,
 					},
 				];
 			}
-		} else if (window.location.pathname.toLowerCase().startsWith("/user:")) {
-			presenceData.state = `Viewing user ${
-				window.location.pathname.match(/\/user:(.*)/i)[1]
-			}`;
-			presenceData.buttons = [
-				{
-					label: "View User",
-					url: window.location.href,
-				},
-			];
-		} else if (
-			window.location.pathname.toLowerCase().startsWith("/category:")
-		) {
-			presenceData.state = `Viewing category ${
-				window.location.pathname.match(/\/category:(.*)/i)[1]
-			}`;
-			presenceData.buttons = [
-				{
-					label: "View Category",
-					url: window.location.href,
-				},
-			];
-		} else if (window.location.pathname.toLowerCase().startsWith("/file:")) {
-			presenceData.state = `Viewing file ${
-				window.location.pathname.match(/\/file:(.*)/i)[1]
-			}`;
-			presenceData.buttons = [
-				{
-					label: "View File",
-					url: window.location.href,
-				},
-			];
+		} else if (pathname.toLowerCase().startsWith("/category:")) {
+			presenceData.state = `Viewing Category: ${urlToTitle(
+				pathname.match(/\/category:(.*)/i)[1]
+			)}`;
+			if (showButtons) {
```

Comment:
Change it to one check at the end.

Like here:
https://github.com/PreMiD/Presences/blob/4f6be8e174deebf7babbcab9384e35d1d9384d38/websites/0-9/5Mods/presence.ts#L165

Reply:
I didn't think about that! I've updated the PR for this!